lms/djangoapps/teams/static/teams/js/views/instructor_tools.js

```diff
@@ -25,14 +29,46 @@
 
                 deleteTeam: function (event) {
                     event.preventDefault();
-                    alert("You clicked the button!");
-                    //placeholder; will route to delete team page
+                    var self = this;
+                    var msg = new PromptView.Warning({
+                        title: gettext('Delete this team?'),
+                        message: gettext('Deleting a team removes it from the team listing view, and removes members from the team as well.'),
+                        actions: {
+                            primary: {
+                                text: gettext('Delete'),
+                                class: 'action-delete',
+                                click: function () {
+                                    msg.hide();
+                                    self.handleDelete();
+                                }
+                            },
+                            secondary: {
+                                text: gettext('Close'),
+                                class: 'action-cancel',
+                                click: function () {
+                                    msg.hide();
+                                }
+                            }
+                        }
+                    });
+                    msg.show()
                 },
 
                 editMembership: function (event) {
                     event.preventDefault();
                     alert("You clicked the button!");
                     //placeholder; will route to remove team member page
+                },
+
+                handleDelete: function () {
+                    var self = this;
+                    this.team.destroy().then(function () {
+                        self.teamEvents.trigger('teams:update', {
```

Comment:
I think you need to update ``collections/topic.js`` so that it marks itself stale on team deletion. It probably deserves a test too, that the team count is correctly updated on the topics page.